### ISSUE TITLE
"Select All Matching" another issue..

### ISSUE THREAD

**Issue by arjasco** [laravel/nova-issues#1356]
- Laravel Version: 5.7.26
- Nova Version: 1.3.2
- PHP Version: 7.2.3
- Operating System and Version: Mac OSX Mojave
- Browser type and version: Chrome 72.0.3626.96

### Description:

When using "Select all matching" with an action the result set is not correct. 

I am trying to collate a CSV export so hundreds even thousands of rows of trivial data, nothing that large.

However as the result set gets larger the collection of models decreases in size at the action.

330 selected has a count of 130 models at the action
450 selected has a count of 50 models at the action.

I need to be able to apply filters also so i can't intervene and do something like the following:

```php
if (request('resources') === 'all') {
    $models = Model::with(...)->get();
}
```

### Steps To Reproduce:

1. Produce a resource over 200 rows
2. Create an action for this resource
3. Assert the count of the collection is not the same as selected on the UI.

![export](https://user-images.githubusercontent.com/8780549/52730715-f0f8cf00-2fb3-11e9-8362-872c812c7622.png)

**Comment by arjasco** [comment]
Seems like it's chunking in 200s and only ending up with the last segment?

**Comment by arjasco** [comment]
Figured it out:

```php
/**
 * The number of models that should be included in each chunk.
 *
 * @var int
 */
public static $chunkCount = 200;
```

🙄

**Comment by dkulyk** [comment]
```php
public function handle(ActionFields $fields, Collection $models)
{
    return $models->count();
}

public function handleResults(ActionFields $fields, $results)
{
    return Action::message(array_sum($results));
}
```

**Comment by arjasco** [comment]
Amazing! thanks @dkulyk  for pointing this out 😄

**Comment by dddeeemmmooonnn** [comment]
Why is it not working?
Created new empty action with only @dkulyk code
![nova_bug1](https://user-images.githubusercontent.com/7344350/52755735-a5ddcb00-304a-11e9-8f58-a92b59d9c380.png)
![nova_bug2 1](https://user-images.githubusercontent.com/7344350/52755737-a6766180-304a-11e9-89d6-a308a0d596ae.png)
![nova_bug3](https://user-images.githubusercontent.com/7344350/52755739-a6766180-304a-11e9-9595-e538eda528b1.png)

**Comment by dkulyk** [comment]
@dddeeemmmooonnn It works since v1.3.2

**Comment by dddeeemmmooonnn** [comment]
@dkulyk it's 1.3.2, I was waiting for the patch because our this

**Comment by dkulyk** [comment]
Покажи код екшена :)

**Comment by dddeeemmmooonnn** [comment]
Ну я с работы уехал уже)
Создал новый экшн, скопировал тот код, всё

**Comment by dkulyk** [comment]
будешь на работе - напиши мне в discord @dkulyk

**Comment by dddeeemmmooonnn** [comment]
не могу в дискорде найти @dkulyk, вот экшн, пустой, с тем кодом

```
<?php

namespace App\Nova\Actions;

use Illuminate\Bus\Queueable;
use Laravel\Nova\Actions\Action;
use Illuminate\Support\Collection;
use Laravel\Nova\Fields\ActionFields;
use Illuminate\Queue\SerializesModels;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Contracts\Queue\ShouldQueue;

class CheckAction extends Action
{
    use InteractsWithQueue, Queueable, SerializesModels;

    /**
     * Perform the action on the given models.
     *
     * @param  \Laravel\Nova\Fields\ActionFields  $fields
     * @param  \Illuminate\Support\Collection  $models
     * @return mixed
     */
    public function handle(ActionFields $fields, Collection $models)
    {
        return $models->count();
    }

    public function handleResults(ActionFields $fields, $results)
    {
        return Action::message(array_sum($results));
    }

    /**
     * Get the fields available on the action.
     *
     * @return array
     */
    public function fields()
    {
        return [];
    }
}
```

**Comment by dkulyk** [comment]
Какая версия Nova?

**Comment by dddeeemmmooonnn** [comment]
Да обновился, 1.3.2, сразу полез это тестить, как ченджлог прочитал.

**Comment by rikless** [comment]
@dddeeemmmooonnn @dkulyk Can you comment in English please ? Pretty hard to follow this trend

**Comment by dkulyk** [comment]
@rikless It works since v1.3.2 :)
@dddeeemmmooonnn у тебя handleResults наследуется с Action ? Если нет - значит ты не обновился.

**Comment by ethanclevenger91** [comment]
@dkulyk `handleResults` should be `handleResult`, I think.

But for anyone confused about this undocumented feature, `handle` gets each chunk, and then `handleResult` gets an array of chunks, which can be merged together and used for any tasks (like an Excel export) that require all the results at once.

**Comment by JoshMoreno** [comment]
Hope this helps

```php
public function handle(ActionFields $fields, Collection $models)
{
    // this runs for every chunk so we just return it so we can
    // merge all the chunks in handleResult below
    return $models;
}

public function handleResult(ActionFields $fields, $results)
{
    /** @var array $results */
    // flatten the chunks into a single collection
    $models = collect($results)->flatten();

    return Action::message($models->count());
}
```

**Comment by philmmoore** [comment]
> Hope this helps
> 
> ```
> public function handle(ActionFields $fields, Collection $models)
> {
>     // this runs for every chunk so we just return it so we can
>     // merge all the chunks in handleResult below
>     return $models;
> }
> 
> public function handleResult(ActionFields $fields, $results)
> {
>     /** @var array $results */
>     // flatten the chunks into a single collection
>     $models = collect($results)->flatten();
> 
>     return Action::message($models->count());
> }
> ```

It's an old one @JoshMoreno but it's just helped me out... thanks!
